xla/service/gpu/tests/dynamic_update_slice_inplace_bf16.hlo

```diff
@@ -0,0 +1,221 @@
+// RUN: hlo-opt %s --platform=gpu --stage=llvm-before-optimizations --xla_gpu_target_config_filename=%S/../../../tools/hlo_opt/gpu_specs/%{GPU}.txtpb --split-input-file | FileCheck --check-prefixes=CHECK %s
+
+// NOTE: Assertions have been autogenerated by utils/generate-test-checks.py
+
+// CHECK-LABEL: @fusion_root_multiple
+// CHECK-SAME: %[[ARG0:[A-Za-z0-9]*]],
+// CHECK-SAME: %[[ARG1:[A-Za-z0-9]*]],
+// CHECK-SAME: %[[ARG2:[A-Za-z0-9]*]],
+// CHECK-SAME: %[[ARG3:[A-Za-z0-9]*]],
+// CHECK-SAME: %[[ARG4:[A-Za-z0-9]*]])
+// CHECK:        %[[COND:[A-Za-z0-9]*]] = icmp ult i64 %[[LINEAR_INDEX:.*]], 132
+// CHECK:        br i1 %[[COND]], label %[[DUS0:.*]].in_bounds-true, label %[[DUS0]].in_bounds-after
+// CHECK:      [[DUS0]].in_bounds-after:
+// CHECK:        %[[COND2:[A-Za-z0-9]*]] = icmp ult i64 %[[LINEAR_INDEX:.*]], 132
+// CHECK:        br i1 %[[COND2]], label %[[DUS1:.*]].in_bounds-true, label %[[DUS1]].in_bounds-after
+// CHECK:      [[DUS1]].in_bounds-after:
+// CHECK-NEXT:   ret void
+// CHECK:      [[DUS0]].in_bounds-true
+// CHECK-DAG:    getelementptr bfloat, ptr %[[ARG1]], i64 %[[VAL_141:.*]]
+// CHECK-DAG:    getelementptr bfloat, ptr %[[ARG3]], i64 %[[VAL_141]]
+// CHECK-DAG:    getelementptr inbounds [10 x [11 x [12 x bfloat]]], ptr %[[ARG0]], i64 0, i64 %[[VAL_185:.*]], i64 %[[VAL_187:.*]], i64 %[[VAL_189:.*]]
+// CHECK:      [[DUS1]].in_bounds-true
+// CHECK-DAG:    getelementptr bfloat, ptr %[[ARG1]], i64 %[[VAL_173:.*]]
+// CHECK-DAG:    getelementptr bfloat, ptr %[[ARG3]], i64 %[[VAL_173]]
+// CHECK-DAG:    getelementptr inbounds [8 x [11 x [12 x bfloat]]], ptr %[[ARG2]], i64 0, i64 %[[VAL_208:.*]], i64 %[[VAL_210:.*]], i64 %[[VAL_212:.*]]
+
+HloModule MultipleInplaceDus, is_scheduled=true, input_output_alias={ {0}: (0, {}), {1}: (2, {}) }
+
+fused_computation {
+  p0 = bf16[10,11,12] parameter(0)
+  p1 = bf16[1,11,12] parameter(1)
+  p2 = bf16[8,11,12] parameter(2)
+  p3 = bf16[1,11,12] parameter(3)
+  p4 = s32[] parameter(4)
+  c0 = s32[] constant(0)
+  cmp = pred[] compare(p4, c0), direction=EQ
+  broadcast = pred[1,11,12] broadcast(cmp), dimensions={}
+  select = bf16[1,11,12] select(broadcast, p1, p3)
+  dus0 = bf16[10,11,12] dynamic-update-slice(p0, select, c0, c0, c0)
+  dus1 = bf16[8,11,12] dynamic-update-slice(p2, select, c0, c0, c0)
+  ROOT tuple = (bf16[10,11,12], bf16[8,11,12]) tuple(dus0, dus1)
+}
+
+ENTRY main {
+  p0 = bf16[10,11,12] parameter(0)
+  p1 = bf16[1,11,12] parameter(1)
+  p2 = bf16[8,11,12] parameter(2)
+  p3 = bf16[1,11,12] parameter(3)
+  p4 = s32[] parameter(4)
+  ROOT fusion_root_multiple = (bf16[10,11,12], bf16[8,11,12]) fusion(p0, p1, p2, p3, p4), kind=kLoop, calls=fused_computation
+}
+
+// -----
+
+// CHECK-LABEL: @fusion_root_multiple_transpose_bitcast
+// CHECK-SAME: %[[ARG0:[A-Za-z0-9]*]],
+// CHECK-SAME: %[[ARG1:[A-Za-z0-9]*]],
+// CHECK-SAME: %[[ARG2:[A-Za-z0-9]*]],
```

Comment:
Do we need to test everything? Maybe just matching a few operations will be enough?

Reply:
fixed